cmd/execution_builder.go

```diff
@@ -934,7 +934,7 @@ func (exeNode *ExecutionNode) LoadFollowerEngine(
 		node.Me,
 		node.Metrics.Engine,
 		node.Storage.Headers,
-		exeNode.builder.FinalizedRootBlock.Header,
+		node.LastFinalizedHeader,
```

Comment:
It was using the finalized root block as the last finalized block, causing all fetch blocks to be rejected.